### PR TITLE
Enable copy elision by supporting symlinks in /pfs/out

### LINKED ISSUE #1784: Copy elision in pipelines
There exists a class of pipelines that simply "shuffles" files, meaning that they simply look at the file names (or maybe read a small part of the files), then move the files into specific directories under `/pfs/out`.

Right now, there's no way for a pipeline to shuffle files without downloading the said files.  This can be prohibitively expensive for certain workloads.

Ideally, there would be a way for a pipeline to copy a file from the input repo to the output repo, without going through the download/upload process.

An idea that @jdoliner proposed was for workers to detect symlinks before uploading the output.  If a file under `/pfs/out` is symlinked to a file in `/pfs/input_repo`, then Pachyderm can be smart enough to not upload the file, but simply to move the file over to the output repo internally.

Combining the approach above with `lazy`, you can have a shuffle pipeline that doesn't download/upload files at all.

### PR DESCRIPTION
**Pull request by derekchiang**
Fix #1784

### CODE REVIEW

File: src/server/pfs/hash.go [hunk 1]
```diff
@@ -1,72 +0,0 @@
...
-
-// BlockInShard returns true if the block is in the given shard.
-func BlockInShard(shard *pfs.Shard, file *pfs.File, block *pfs.Block) bool {
-	if shard == nil || shard.BlockModulus == 0 {
-		// this lets us default to no filtering
-		return true
-	}
-	sharder := &Hasher{BlockModulus: shard.BlockModulus}
-	return sharder.HashBlock(file, block) == shard.BlockNumber
-}
```
Comment (jdoliner) [inline src/server/pfs/hash.go]: This was my favorite code though :(

File: src/server/pps/server/api_server.go [hunk 2]
```diff
@@ -1792,6 +1792,7 @@ func (a *apiServer) jobManager(ctx context.Context, jobInfo *pps.JobInfo) {
 		}()
 		clientPool := sync.Pool{
 			New: func() interface{} {
+				protolion.Infof("Creating a new worker connection for job %s", jobID)
```
Comment (jdoliner) [inline src/server/pps/server/api_server.go]: Not sure you meant to commit this.

**Comment by jdoliner** [comment]
1 comment you may wanted to address about a lingering `Printf` other than that looks good to me.

File: src/server/pfs/hash.go (hunk 1 above)
Comment (derekchiang) [inline src/server/pfs/hash.go]: em memories

**Pull request merged**